client/document.go

```diff
@@ -119,15 +117,274 @@ func NewDocFromMap(data map[string]any) (*Document, error) {
 	return doc, nil
 }
 
+var jsonArrayPattern = regexp.MustCompile(`^\s*\[.*\]\s*$`)
+
+// IsJSONArray returns true if the given byte array is a JSON Array.
+func IsJSONArray(obj []byte) bool {
+	return jsonArrayPattern.Match(obj)
+}
+
 // NewFromJSON creates a new instance of a Document from a raw JSON object byte array.
-func NewDocFromJSON(obj []byte) (*Document, error) {
-	data := make(map[string]any)
-	err := json.Unmarshal(obj, &data)
+func NewDocFromJSON(obj []byte, sd SchemaDescription) (*Document, error) {
+	doc := newEmptyDoc(sd)
+	err := doc.SetWithJSON(obj)
 	if err != nil {
 		return nil, err
 	}
+	err = doc.generateAndSetDocKey()
+	if err != nil {
+		return nil, err
+	}
+	return doc, nil
+}
+
+// ManyFromJSON creates a new slice of Documents from a raw JSON array byte array.
+func NewDocsFromJSON(obj []byte, sd SchemaDescription) ([]*Document, error) {
```

Comment:
question: What behaviour will this function when given a non-array (e.g. a single doc, as would be given to `NewDocFromJSON`?

todo: Please document this (public) behaviour in the func doc, and with at least one test.

Reply:
It will error with `value doesn't contain array; it contains object`.

Reply:
✅